### PR TITLE
Deprecate torch.matrix_power in favor of torch.linalg.matrix_power

### PR DESCRIPTION
**Pull request by heitorschueroff**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53538 Deprecate torch.matrix_power in favor of torch.linalg.matrix_power**
* #52608 Added torch.linalg.matrix_power

Differential Revision: [D27261531](https://our.internmc.facebook.com/intern/diff/D27261531)

### CODE REVIEW

**Review by mruberry — COMMENTED** [review]
What should we do about
 https://github.com/pytorch/pytorch/blob/a76b4736dbf6c91f11e57c2141104452d0c2fc4c/test/test_linalg.py#L5396

?

Should it be removed? Or updated to call linalg.matrix_power()?

**Comment by mruberry** [comment]
I think we can make torch.matrix_power() an alias of torch.linalg.matrix_power()? We have support for aliases in OpInfos, too. And then the docs can point torch.matrix_power() to torch.linalg.matrix_power().

**Comment by heitorschueroff** [comment]
> What should we do about
> https://github.com/pytorch/pytorch/blob/a76b4736dbf6c91f11e57c2141104452d0c2fc4c/test/test_linalg.py#L5396
> 
> ?
> 
> Should it be removed? Or updated to call linalg.matrix_power()?

I removed it because the new functions cover the same tests and more.

**Comment by heitorschueroff** [comment]
@mruberry I added the out parameter to `torch.matrix_power` and made it into an alias. I fixed some bugs with the alias testing as well.

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -2270,11 +2270,12 @@ def callable(a, b) -> number
 See :func:`torch.masked_select`
 """)
 
-add_docstr_all('matrix_power',
-               r"""
+add_docstr_all('matrix_power', r"""
 matrix_power(n) -> Tensor
 
-See :func:`torch.matrix_power`
+.. note:: :meth:`~Tensor.matrix_power` is deprecated, use :func:`torch.linalg.matrix_power` instead.
```
Comment (mruberry) [inline torch/_tensor_docs.py:2276]: This can still just point to the torch.matrix_power docs like the other methods

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -5066,37 +5066,12 @@ def merge_dicts(*dicts):
...
-            [[-1.2534, -1.3429],
-             [ 0.4153, -1.4664]]])
-    >>> torch.matrix_power(a, 3)
-    tensor([[[  3.9392, -23.9916],
-             [ 11.7357,  -0.2070]],
+.. note:: :func:`torch.matrix_power` is deprecated, use :func:`torch.linalg.matrix_power` instead.
 
-            [[  0.2468,  -6.7168],
-             [  2.0774,  -0.8187]]])
+Alias for :func:`torch.linalg.matrix_power`
```
Comment (mruberry) [inline torch/_torch_docs.py:5074]: Period at end of this sentence (yes it's super weird that the convention in _torch_docs is not to put a period at the end of its reference sentence; I don't know how that got started).

**Review by mruberry — APPROVED** [review]
Cool! Just two minor docs suggestions.

The merge conflicts are odd and unlikely to be related to your changes. I've hit similar issues in the past, too. Importing, rebasing in jellyfish, and exporting might get the PR CI going.

**Pull request closed without merging**